---
title: C6255
ms.date: 11/04/2016
ms.prod: visual-studio-dev15
ms.topic: reference
f1_keywords:
- C6255
helpviewer_keywords:
- C6255
ms.assetid: bb6430b2-782a-4410-a8e1-609df06007de
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 3c2edfc5bbf3ac27ab36f39d002cdcf4383b1482
ms.sourcegitcommit: 37fb7075b0a65d2add3b137a5230767aa3266c74
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 01/02/2019
ms.locfileid: "53928566"
---
# <a name="c6255"></a>C6255
Avertissement C6255 : _alloca signale un échec en levant une exception de dépassement de capacité de pile. Utilisez _malloca à la place

 Cet avertissement indique qu’un appel à `_alloca` a été détecté en dehors de la gestion des exceptions locales. `_alloca` doit toujours être appelée à partir de la plage protégée d’un gestionnaire d’exceptions car il peut lever une exception de dépassement de capacité de pile en cas d’échec. Si possible, au lieu d’utiliser `_alloca`, envisagez d’utiliser `_malloca` qui est une version plus sécurisée de `_alloca`.

## <a name="example"></a>Exemple
 Le code suivant génère cet avertissement car `_alloca` peut générer l’exception :

```
#include <windows.h>

void f( )
{
  void *p = _alloca(10);
  // code ...
}
```

 Pour corriger cet avertissement, utilisez `_malloca` et ajoutez le Gestionnaire d’exceptions comme indiqué dans le code suivant :

```
#include <windows.h>
#include <malloc.h>

void f( )
{
  void *p;
  int errcode;
  __try
  {
    p = _malloca(10);
    // code...
    _freea(p);
  }
  __except( (GetExceptionCode() == STATUS_STACK_OVERFLOW ) ?
              EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )
  {
    errcode = _resetstkoflw();
    // code ...
  }
}
```

## <a name="see-also"></a>Voir aussi
 [_malloca](/cpp/c-runtime-library/reference/malloca)